components/post_view/message_attachments/message_attachment/message_attachment.jsx

```diff
@@ -67,14 +67,24 @@ export default class MessageAttachment extends React.PureComponent {
         };
     }
 
+    componentDidMount() {
+        this.mounted = true;
+    }
+
+    componentWillUnmount() {
+        this.mounted = false;
+    }
+
     handleHeightReceived = (height) => {
         if (height > 0) {
             // Increment checkOverflow to indicate change in height
             // and recompute textContainer height at ShowMore component
             // and see whether overflow text of show more/less is necessary or not.
-            this.setState((prevState) => {
-                return {checkOverflow: prevState.checkOverflow + 1};
-            });
+            if (this.mounted) {
+                this.setState((prevState) => {
```

Comment:
@sudheerDev, here's an example of what I mean: should we just check `this.mounted` at the start of the function and do no other work in the handler?

Reply:
I will go though through this PR tomorrow for sure and apply you suggestion of bailing out not just setState but any other code execution if possible.

Lemme google this again to see if there are any new patterns i can find about this. last time i did there was just few stackoverflow and github references mostly doing pretty much the same

Reply:
Aah yes there were a lot of instances of imageLoad as well. I didn't go though it when i was replying earlier. I removed them earlier because of post metadata but possibly added few back to have backwards compatibility.

I wouldn't worry too much about them as we will be removing them again

Reply:
@lieut-data we have three type of actions which cause these async setState calls

1. images loaded with new Image() which can be cancelled by changing src and image.onload to an empty function which is similar to cancelling

2. images request with XMLHttpRequest which can be canelled with abort()

3. async API calls (We dont have a way to cancel these as of now)

i thought changing first two cases for now but ran into an issue with the utility function we use in few places. 

https://github.com/mattermost/mattermost-webapp/blob/279ae94907fad14020a79fa960f3525260eb644a/utils/utils.jsx#L1146-L1161

the utility function does not return anything which might be causing couple of bugs so leaving with this pattern for now.

I am inclined on creating a ticket for investigation to remove  `mounted pattern` and come up with cancel patterns for different scenarios.